presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/BasePickTableLayoutTest.java

```diff
@@ -182,4 +189,20 @@ public void ruleAddedNewTableLayoutIfTableScanHasEmptyConstraint()
                                         ImmutableMap.of("nationkey", singleValue(BIGINT, 44L)),
                                         ImmutableMap.of("nationkey", "nationkey"))));
     }
+
+    @Test
+    public void ruleWithPushdownableToTableLayoutPredicate()
```

Comment:
Can you squash this with code changes in the rule itself?

Reply:
Too bad. I've squashed that many commits in CBO already. I vehemently reject a rule preventing squashing commits of different authors (regardless of whether _my_ commits is first, second or none).

Reply:
This requires all these commits to be squashed together:
```
pick ed57df9636 Execute PickTableLayout before AddExchanges
pick 3c90bd29c5 Test PickTableLayout with predicate pushdown enabled TPCH connector
pick 689b9e0b58 Add a pick layout rule test where predicate is fully enforced
```

Let it be.